python/lsst/pex/config/config.py

```diff
@@ -456,13 +444,8 @@ def _parseTypingArgs(
             _typ = ForwardRef(unpackedParams)
             # type ignore below because typeshed seems to be wrong. It
             # indicates there are only 2 args, as it was in python 3.8, but
-            # 3.9+ takes 3 args. Attempt in old style and new style to
-            # work with both.
-            try:
-                result = _typ._evaluate(globals(), locals(), set())  # type: ignore
-            except TypeError:
-                # python 3.8 path
-                result = _typ._evaluate(globals(), locals())
+            # 3.9+ takes 3 args.
+            result = _typ._evaluate(globals(), locals(), recursive_guard=set())  # type: ignore
```

Comment:
@natelust it seems that python 3.12 changed the type signature for this internal method and you need to explicitly use `recursive_guard=`.

Reply:
Sounds fine to me. I was reading ahead, and in python 3.13 there is a way to do all this much more appropriately though apis, so I am looking forward to that.

Reply:
The interesting thing is that python 3.12 did work previously so a minor patch release broke it (which I imagine they don't care about since it's an internal API).